langchain-core/src/utils/signal.ts

```diff
@@ -14,13 +15,14 @@ export async function raceWithSignal<T>(
       }
     }),
     new Promise<never>((_, reject) => {
-      signal.addEventListener("abort", () => {
+      listener = () => {
         reject(new Error("Aborted"));
```

Comment:
I think the remove call should be inside the listener?

Reply:
Could add it in both places but it should be removed once the promise resolves too?

What's the benefit to putting it within the listener vs. in `finally`?